### PR TITLE
Forbid Pydantic >= 2 for napari 0.4.17

### PR DESCRIPTION
**Pull request by jni**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] ~~Reset the build number to `0` (if the version changed)~~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

See napari/napari#6031 napari/napari#6029

### CODE REVIEW

**Comment by jni** [comment]
@conda-forge-admin please rerender

**Comment by jni** [comment]
@conda-forge-admin, please rerender

**Comment by jaimergp** [comment]
This kind of task is usually better solved by a repodata patch. Having it on the feedstock too doesn't hurt so the next release inherits it, but we definitely need the repo patch. Can you open one and we'll merge it quickly?

Some examples:

* https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/476
* https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/475

File: recipe/meta.yaml [hunk 1]
```diff
@@ -60,7 +60,7 @@ outputs:
         - pint >=0.17
         - psutil >=5.0
         - psygnal>=0.3.4
-        - pydantic >=1.9.0
+        - pydantic >=1.9.0,<2
```
Comment (jaimergp) [inline recipe/meta.yaml:63]: Should have been `<2.0a0` 😬

**Comment by jaimergp** [comment]
Went ahead and posted https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/477

File: recipe/meta.yaml (hunk 1 above)
Comment (jni) [reply recipe/meta.yaml:63]: Wow really??? 🤦

**Pull request merged**